src/cursor.coffee

```diff
@@ -304,13 +304,14 @@ class Cursor extends Model
     else
       {row, column} = @getScreenPosition()
 
+      originalRow = row
       while columnCount > column and row > 0
         columnCount -= column
         column = @editor.lineTextForScreenRow(--row).length
         columnCount-- # subtract 1 for the row move
 
       column = column - columnCount
-      @setScreenPosition({row, column})
+      @setScreenPosition({row, column}, wrapAtSoftNewlines: originalRow == row)
```

Comment:
I'm trying a commit where `wrapAtSoftNewlines` and `skipAtomicTokens` are `true` when moving down. It probably should have been that way anyway.